src/cpp-simulator/qiskitsimulator.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2017 IBM RESEARCH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+"""
+Interface to Realistic Noise Simulator.
+
+Date: Nov 29 2017
+Author: Christopher J. Wood <cjwood@us.ibm.com>
+"""
+
+import os
+import json
+import subprocess
+from subprocess import PIPE, CalledProcessError
+import numpy as np
+import numbers
+
+
+def simulator_config():
+    """
+    Return the configuration dictionary for local_qiskit_simulator.
+    """
+    return {
+        "name": "local_qiskit_simulator",
+        "url": "https://github.com/QISKit/qiskit-sdk-py/external/cpp-simulator",
+        "simulator": True,
+        'local': True,
+        "description": "A C++ realistic noise simulator for qobj files",
+        "coupling_map": "all-to-all",
+        "basis_gates": "u1,u2,u3,cx,id,x,y,z,h,s,sdg,t,tdg,wait,noise,save,load,uzz"
+    }
+
+
+def clifford_config():
+    """
+    Return the configuration dictionary for local_clifford_simulator.
+    """
+    return {
+        "name": "local_clifford_simulator",
+        "url": "https://github.com/QISKit/qiskit-sdk-py/external/cpp-simulator",
+        "simulator": True,
+        'local': True,
+        "description": "A C++ Clifford simulator with approximate noise",
+        "coupling_map": "all-to-all",
+        "basis_gates": "cx,id,x,y,z,h,s,sdg,wait,noise,save,load"
+    }
+
+
+def run(qobj, exe='local_qiskit_simulator', path=None):
+    """
+    Run simulation on C++ simulator.
+
+    Args:
+        qobj (dict): qobj dictionary defining the simulation to run
+        exe (str): qobj simulator executable (default: 'wood_simulator')
+        path (str): path to exe directory (default: None)
+
+    Returns:
+        A dict of simulation results
+    """
+    # Get path to exe
+    if path is None:
+        cmd = os.path.dirname(os.path.abspath(__file__))
+    else:
+        cmd = os.path.expanduser(path)
+    # Add exe name
+    cmd = os.path.join(cmd, exe)
```

Comment:
I'm a bit concerned about the security implications of this. If the user can specify where the cpp simulator binary is located, this will open a window for executing untrusted code in the system. I'm gonna review this carefully and see the real implications.